packages/fuels-programs/src/contract.rs

```diff
@@ -45,6 +84,160 @@ impl Contract {
+    ) -> Result<Self> {
+        let code = Self::loader_contract(blob_ids)?;
+        Ok(Self::new(code, salt, storage_slots))
+    }
+
+    /// Splits the contract into blobs, submits them, and awaits confirmation. Then, it deploys a loader contract.
+    /// This loader contract will load the blobs into memory and delegate the call to the code contained within the blobs.
+    /// This method is useful for deploying large contracts.
+    pub async fn deploy_as_loader(
+        self,
+        account: &impl Account,
+        tx_policies: TxPolicies,
+        blob_size_policy: BlobSizePolicy,
+    ) -> Result<Bech32ContractId> {
+        let provider = account.try_provider()?;
+
+        let blobs = self.generate_blobs(provider, blob_size_policy).await?;
+        let blob_ids = blobs.iter().map(|blob| blob.id()).collect::<Vec<_>>();
+
+        for blob in blobs {
+            let mut tb = BlobTransactionBuilder::default()
+                .with_blob(blob)
+                .with_tx_policies(tx_policies)
+                .with_max_fee_estimation_tolerance(0.05);
+
+            account.adjust_for_fee(&mut tb, 0).await?;
+            account.add_witnesses(&mut tb)?;
+
+            let tx = tb.build(provider).await?;
+            provider
+                .send_transaction_and_await_commit(tx)
+                .await?
+                .check(None)?;
+        }
+
+        Self::new_loader(&blob_ids, self.salt, self.storage_slots)?
+            .deploy(account, tx_policies)
+            .await
+    }
+
+    /// Splits the contract binary into blobs based on the size specified by `blob_size_policy`.
+    /// This is useful if you prefer to manually deploy the blobs. Once uploaded, you can use [`Contract::new_loader`] to create a loader contract.
+    pub async fn generate_blobs(
+        &self,
+        provider: &Provider,
+        policy: BlobSizePolicy,
+    ) -> Result<Vec<Blob>> {
+        let blob_size = policy.resolve_size(provider).await?;
+
+        let blobs = self
+            .binary
+            .chunks(blob_size)
+            .map(|chunk| Blob::new(chunk.to_vec()))
+            .collect();
+
+        Ok(blobs)
+    }
+
+    // This function creates a contract that loads the specified blobs into memory and delegates the call to the code contained in the blobs.
+    fn loader_contract(blob_ids: &[[u8; 32]]) -> Result<Vec<u8>> {
+        const BLOB_ID_SIZE: u16 = 32;
+        let get_instructions = |num_of_instructions, num_of_blobs| {
+            // There are main 3 steps:
+            // 1. Calculate the total size of the contract
+            // 2. Allocate and load the contract into memory
+            // 3. Jump into the memory where the contract is loaded
+            // After that the execution continues normally with the loaded contract reading our
+            // prepared fn selector and jumps to the selected contract method.
+            [
+                // 0x12 is going to hold the total size of the contract
```

Comment:
To make this ASM block maintainable I would add where each of the 3 steps is starting.
```suggestion
                // 1. Calculate the total size of the contract
                // 0x12 is going to hold the total size of the contract
```